runtime/gc_trace/TgcParallel.cpp

```diff
@@ -305,7 +305,7 @@ tgcHookLocalGcEnd(J9HookInterface** hook, uintptr_t eventNum, void* eventData, v
 	uint64_t prevReadObjectBarrierUpdate = 0;
 #endif /* OMR_GC_CONCURRENT_SCAVENGER */
 	while (historyRecord < endRecord) {
-		totalRecordUpdates += historyRecord->updates;
+		totalRecordUpdates += (uintptr_t) historyRecord->updates;
```

Comment:
Why the mix of different size types (`uintptr_t` and `uint64_t`)? Why not just `uint64_t` throughout?

Reply:
These counts should all be `uintptr_t. They'll need to be all aligned to to use the same type, see https://github.com/eclipse/omr/issues/4825#issuecomment-589370296

Reply:
These counts should all be `uintptr_t`. They'll need to be all aligned to to use the same type, see https://github.com/eclipse/omr/issues/4825#issuecomment-589370296.

Requires changes on the OMR side, will need to get PR in before making the counts consistent

Reply:
If everything should be `uintptr_t` why is `uint64_t` used in this function (e.g. `totalRecordUpdates`)?

Reply:
There's 3 different variables here:
-  `uint64_t` `historyRecord->updates`  
- `uintptr_t` `extensions->scavengerStats._copyScanUpdates` 
- `uintptr_t` `totalRecordUpdates`

There's no way to get around casting without any changes to OMR side. If we forgo casting 
`historyRecord->updates` here to `uintptr_t`, we'd have to change `totalRecordUpdates` `uint64_t` to match `historyRecord->updates`.... however then we'd need to cast `extensions->scavengerStats._copyScanUpdates`  to `uint64_t` a few lines down when we subtract `totalRecordUpdates` and `extensions->scavengerStats._copyScanUpdates`:
https://github.com/eclipse/openj9/blob/8782135e7f969dacae908af4c7c0d68511a94174/runtime/gc_trace/TgcParallel.cpp#L340

Reply:
I hope we get this sorted out soon; casts are just a way to hide real problems.